### PR TITLE
[NFC] Define InheritedProtocolsRequest

### PR DESCRIPTION
**Pull request by CodaFi**
Refactor the interface to ProtocolDecl::getInheritedProtocols in
preparation for request-based dependency tracking.

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci please test

**Review by hamishknight — APPROVED** [review]

File: lib/AST/NameLookup.cpp [hunk 1]
```diff
@@ -2235,6 +2235,23 @@ SuperclassDeclRequest::evaluate(Evaluator &evaluator,
   return nullptr;
 }
 
+ArrayRef<ProtocolDecl *>
+InheritedProtocolsRequest::evaluate(Evaluator &evaluator,
+                                    ProtocolDecl *PD) const {
+  llvm::SmallVector<ProtocolDecl *, 2> result;
+  SmallPtrSet<const ProtocolDecl *, 2> known;
+  known.insert(PD);
+  bool anyObject = false;
```
Comment (slavapestov) [inline lib/AST/NameLookup.cpp]: This is doing the same calculation as ProtocolRequiresClassRequest, except it's keeping a different part of the result. Perhaps we can combine the two into a single request that evaluates into an std::pair<>, and then ::requiresClass() and ::getInheritedProtocols() can kick off the combined request and project a different component of the result?

**Review by slavapestov — APPROVED** [review]

**Comment by CodaFi** [comment]
Builder died

@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please test macOS platform

**Comment by CodaFi** [comment]
@swift-ci smoke test macOS platform

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**